scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2142,34 +2147,63 @@ def __init__(self, k8s, port=10255, host_ip=None):
         self._session = requests.Session()
         headers = {
             "Accept": "application/json",
+            "Authorization": "Bearer %s" % k8s.token,
         }
         self._session.headers.update(headers)
 
-        global_log.info("KubeletApi host ip = %s" % host_ip)
-        if host_ip:
-            self._http_host = "http://%s:%d" % (host_ip, port)
-        else:
-            self._http_host = None
+        global_log.info("KubeletApi host ip = %s" % self._host_ip)
+        self._build_host()
         self._timeout = 20.0
 
+    def _build_host(self):
```

Comment:
Replace this with `_build_kubelet_url` mentioned above.  Also make it static and add a test case to verify the string.Template substitution works correctly.